File: An2/MPP/mppweb/lib/models/categorievarsta.dart
import 'entitate.dart';
import 'dart:convert';


class CategorieVarsta extends Entitate<int> {
  int varstaMinima;
  int varstaMaxima;

  CategorieVarsta({required int id,required this.varstaMinima,required this.varstaMaxima}) : super(id: id);

  factory CategorieVarsta.fromJson(Map<String, dynamic> json) {
    return CategorieVarsta(
      id: json['id'],
      varstaMinima: json['varstaMinima'],
      varstaMaxima: json['varstaMaxima'],
    );
  }

  Map<String, dynamic> toJson() => {
    'id': id,
    'varstaMinima': varstaMinima,
    'varstaMaxima': varstaMaxima,
  };
  String toString() {
    return varstaMinima.toString() + '-' + varstaMaxima.toString();
  }
}

File: An2/MPP/mpp-proiect-java-DeepSmeag/modelModule/src/main/java/contests/model/CategorieVarsta.java
package contests.model;


import contests.model.entity.Entitate;

public class CategorieVarsta extends Entitate<Integer> {
    private int varstaMinima;
    private int varstaMaxima;

    public CategorieVarsta() {}
    public CategorieVarsta(Integer id, int varstaMinima, int varstaMaxima) {
        super(id);
        this.varstaMinima = varstaMinima;
        this.varstaMaxima = varstaMaxima;
    }

    public int getVarstaMinima() {
        return varstaMinima;
    }

    public int getVarstaMaxima() {
        return varstaMaxima;
    }

    public void setVarstaMinima(int varstaMinima) {
        this.varstaMinima = varstaMinima;
    }

    public void setVarstaMaxima(int varstaMaxima) {
        this.varstaMaxima = varstaMaxima;
    }

    public String toString() {
        return varstaMinima + "-" + varstaMaxima;
    }
}
